keps/sig-apps/4563-evacuation-api/README.md

```diff
@@ -0,0 +1,1194 @@
+   the minimum number of pods is 1, PDBs cannot be used without blocking the node drain. This has
+   been discussed in issue [kubernetes/kubernetes#93476](https://github.com/kubernetes/kubernetes/issues/93476).
+3. Graceful deletion of DaemonSet pods is currently only supported as part of (Linux) graceful node
+   shutdown. The length of the shutdown is again not application specific and is set cluster-wide
+   (optionally by priority) by the cluster admin. This does not take into account
+   `.spec.terminationGracePeriodSeconds` of each pod and may cause premature termination of
+   the application. This has been discussed in issue [kubernetes/kubernetes#75482](https://github.com/kubernetes/kubernetes/issues/75482)
+   and in issue [kubernetes-sigs/cluster-api#6158](https://github.com/kubernetes-sigs/cluster-api/issues/6158).
+4. Descheduler does not allow postponing eviction for applications that are unable to be evicted
+   immediately. This can result in descheduling of incorrect set of pods. This is outlined in the
+   KEP [kubernetes-sigs/descheduler#1354](https://github.com/kubernetes-sigs/descheduler/pull/1354).
+
+### Goals
+
+- Introduce new `evacuation.coordination.k8s.io` API.
+- Introduce evacuation controller.
+
+### Non-Goals
+
+- Synchronizing of the evacuation status to the pod status.
+
+## Proposal
+
+We will introduce a new term called evacuation. This is a contract between the evacuation instigator,
+the evacuee, and the evacuator. The contract is enforced by the API and an evacuation controller.
+We can think of evacuation as a managed and safer alternative to eviction.
+
+### Evacuation Instigator
+
+The evacuation instigator can be any entity in the system: node maintenance controller, descheduler,
+cluster autoscaler, or any application/controller interfacing with the affected application/pods
+(evacuee).
+
+The instigator's responsibility is to communicate an intent to a pod that it should be evacuated,
+according to the instigator's own internal rules. It should reconcile its intent in case the intent
+is removed by a third party. And it should remove its intent when the evacuation is no longer necessary.
+
+Example evacuation triggers:
+- Node maintenance controller: node maintenance triggered by an admin.
+- Descheduler: descheduling triggered by a descheduling rule.
+- Cluster autoscaler: node downscaling triggered by a low node utilization.
+
+It is understood that multiple evacuation instigators may request evacuation of the same pod at the
+same time. The instigators should coordinate their intent and not remove the evacuation until all
+instigators have dropped their intent.
+
+### Evacuee and Evacuator
+
+The evacuee can be any pod. The owner/controller of the pod is the evacuator. In a
+special case, the evacuee can be its own evacuator. The evacuator should decide what action to take
+when it observes an evacuation intent:
```

Comment:
This doesn't match common English use. An evacuator would be something initiating evacuation, not the owner / manager of the thing being evactuated.

Reply:
I don't know: It seemed to fit somehow, according to the [dictionary](https://www.dictionary.com/browse/evacuator) ^^ 
> a person or thing that evacuates.

In my view, even though it is the owner/manager of the pod, it also implements the evacuation part (evacuates).

Nevertheless, I  am open to suggestions.

How about `Evacuator -> Evacuation Responder -> Evacuee`?